internal/cmd/scraper/scraper.go

```diff
@@ -116,12 +116,12 @@ func RunWithEmitters(cfg *Config, emitters []integration.Emitter) error {
 		return fmt.Errorf("you need to configure at least one valid emitter")
 	}
 
```

Comment:
If we added ctx context to this RunWithEmitters and allowed the server creation to be done creting the server instead of calling the helper function [listenAndServe](https://cs.opensource.google/go/go/+/refs/tags/go1.18:src/net/http/server.go;drc=b5e06545b3c3b3b9c2ed8204e5ba3fd25a85db0f;l=3220) I would be able to put a go routine listening to the context cancel and forcing the server to shutdown, so we can create a test like the one there is for RunOnceWithEmitters, but I understand that, since we are thinking on refactoring all the integration, it's better to touch the less as posible and just test this manually by running the binary, do you agree @gsanchezgavier ?